command/connect/envoy/envoy_test.go

```diff
@@ -527,3 +528,99 @@ func testMockAgentProxyConfig(cfg map[string]interface{}) http.HandlerFunc {
 		w.Write(cfgJSON)
 	})
 }
+
+func TestEnvoyCommand_canBindInternal(t *testing.T) {
+	t.Parallel()
+	type testCheck struct {
+		expected bool
+		addr     string
+	}
+
+	type testCase struct {
+		ifAddrs []net.Addr
+		checks  map[string]testCheck
+	}
+
+	parseIPNets := func(t *testing.T, in ...string) []net.Addr {
+		var out []net.Addr
+		for _, addr := range in {
+			ip := net.ParseIP(addr)
+			require.NotNil(t, ip)
+			out = append(out, &net.IPNet{IP: ip})
+		}
+		return out
+	}
+
+	parseIPs := func(t *testing.T, in ...string) []net.Addr {
+		var out []net.Addr
+		for _, addr := range in {
+			ip := net.ParseIP(addr)
+			require.NotNil(t, ip)
+			out = append(out, &net.IPAddr{IP: ip})
+		}
+		return out
+	}
+
+	cases := map[string]testCase{
+		"IPNet": {
+			parseIPNets(t, "10.3.0.2", "198.18.0.1", "2001:db8:a0b:12f0::1"),
```

Comment:
shouldn't that be something else? `IPNet` and `IPAddr` look the same.

Reply:
The only difference between parseIPNets and parseIPs is that the first returns as `[]*net.IPNet` and the second `[]*net.IPAddr` (which both are then are just `[]net.Addr`.

The type switch in canBind mainly is looking for `net.IPNet` as that is the concrete datatype currently returned by `net.InterfaceAddrs` however the default case which gets used by the `[]*net.IPAddr` handles the case where other implementors of `net.Addr` are returned instead. So the only difference in the tests are the concrete data type but I wanted to ensure that both worked.

Reply:
As a side note, it seems kind of crazy that `net.InterfaceAddrs` doesn't just return a `[]*net.IPNet` instead of using the `net.Addr` interface. I don't like not knowing what the real type to expect is.